### PR TITLE
Fix `Exp10M1(TSelf)` and `Round(TSelf, MidpointRounding)`

### LINKED ISSUE #75059: Wrong DIM for `Exp10M1(TSelf)` and `Round(TSelf, MidpointRounding)`
### Description

DIM for  `Exp10M1(TSelf)` causes recursion
DIM for `Round(TSelf, MidpointRounding)` ignored second parametr

### Reproduction Steps

Just look at the code [here](https://github.com/dotnet/runtime/blob/e71a9583b4d6c9bd97edd87cda7f98f232f63530/src/libraries/System.Private.CoreLib/src/System/Numerics/IExponentialFunctions.cs#L40) and [here](https://github.com/dotnet/runtime/blob/ca82565a60380bf4220255c65e493deb44314346/src/libraries/System.Private.CoreLib/src/System/Numerics/IFloatingPoint.cs#L39)

### Expected behavior

[Exp10M1(TSelf x)](https://github.com/dotnet/runtime/blob/e71a9583b4d6c9bd97edd87cda7f98f232f63530/src/libraries/System.Private.CoreLib/src/System/Numerics/IExponentialFunctions.cs#L40): get the result
[Round(TSelf x, MidpointRounding mode)](https://github.com/dotnet/runtime/blob/ca82565a60380bf4220255c65e493deb44314346/src/libraries/System.Private.CoreLib/src/System/Numerics/IFloatingPoint.cs#L39): parameter `mode`will not be ignored

### Actual behavior

[Exp10M1(TSelf x)](https://github.com/dotnet/runtime/blob/e71a9583b4d6c9bd97edd87cda7f98f232f63530/src/libraries/System.Private.CoreLib/src/System/Numerics/IExponentialFunctions.cs#L40): StackOverflowException
[Round(TSelf x, MidpointRounding mode)](https://github.com/dotnet/runtime/blob/ca82565a60380bf4220255c65e493deb44314346/src/libraries/System.Private.CoreLib/src/System/Numerics/IFloatingPoint.cs#L39): parameter `mode` is ignored

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by blokyk**
`IExponentialFunctions.Exp10M1` had a typo that caused it to go into infinite recursion, and `IFloatingPoint.Round` straight up just ignored it's `mode` argument, most likely just a typo again.

Fixes #75059

### CODE REVIEW

**Comment by Blokyk** [comment]
I apologize for the lack of tests -- I'm currently experiencing connectivity issues, which makes cloning and building the runtime a bit tricky. I'm writing some right now for both of the functions that I fixed, but if needed I can try to add more while I wait for my internet to come back. Sorry for submitting this PR without the tests first

**Comment by tannergooding** [comment]
> I apologize for the lack of tests -- I'm currently experiencing connectivity issues, which makes cloning and building the runtime a bit tricky. I'm writing some right now for both of the functions that I fixed, but if needed I can try to add more while I wait for my internet to come back. Sorry for submitting this PR without the tests first

Absolutely no worries. We'd much rather have a fix up than not at all. Someone else can always pick it up and add tests if necessary/etc 😄

File: src/libraries/System.Runtime/tests/System/Numerics/GenericMathDimHelpers.cs [hunk 1]
```diff
@@ -96,4 +96,177 @@ private BinaryNumberDimHelper(int value)
...
+    {
+        public float Value;
+
+        public ExponentialFunctionsDimHelper(float value)
+        {
+            Value = value;
+        }
+
+
+        static ExponentialFunctionsDimHelper IExponentialFunctions<ExponentialFunctionsDimHelper>.Exp10(ExponentialFunctionsDimHelper x) => new ExponentialFunctionsDimHelper(Single.Exp10(x.Value));
```
Comment (tannergooding) [inline src/libraries/System.Runtime/tests/System/Numerics/GenericMathDimHelpers.cs:203]: Same nit:
```suggestion
        static ExponentialFunctionsDimHelper IExponentialFunctions<ExponentialFunctionsDimHelper>.Exp10(ExponentialFunctionsDimHelper x) => new ExponentialFunctionsDimHelper(float.Exp10(x.Value));
```

File: src/libraries/System.Runtime/tests/System/Numerics/GenericMathDimHelpers.cs [hunk 2]
```diff
@@ -96,4 +96,177 @@ private BinaryNumberDimHelper(int value)
...
+    public struct ExponentialFunctionsDimHelper : IExponentialFunctions<ExponentialFunctionsDimHelper>
+    {
+        public float Value;
+
+        public ExponentialFunctionsDimHelper(float value)
+        {
+            Value = value;
+        }
+
+
```
Comment (tannergooding) [inline src/libraries/System.Runtime/tests/System/Numerics/GenericMathDimHelpers.cs:202]: nit: stray newline
```suggestion
```

File: src/libraries/System.Runtime/tests/System/Numerics/IExponentialFunctionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public sealed class IExponentialFunctionsTests
+    {
+        const float baseValue = 2.1f;
+        static readonly ExponentialFunctionsDimHelper helperValue = new ExponentialFunctionsDimHelper(baseValue);
+
+        [Fact]
+        public static void Exp10M1Test()
+        {
+            Assert.Equal(Single.Exp10M1(baseValue), ExponentialFunctionsHelper<ExponentialFunctionsDimHelper>.Exp10M1(helperValue).Value);
```
Comment (tannergooding) [inline src/libraries/System.Runtime/tests/System/Numerics/IExponentialFunctionsTests.cs:16]:
```suggestion
            Assert.Equal(float.Exp10M1(baseValue), ExponentialFunctionsHelper<ExponentialFunctionsDimHelper>.Exp10M1(helperValue).Value);
```

File: src/libraries/System.Runtime/tests/System/Numerics/IFloatingPointTests.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+{
+    public sealed class IFloatingPointTests
+    {
+        const float baseValue = 5.5f;
+        static readonly FloatingPointDimHelper helperValue = new FloatingPointDimHelper(baseValue);
+
+        [Fact]
+        public static void AwayFromZeroRoundingTest()
+        {
+            Assert.Equal(Single.Round(baseValue, MidpointRounding.AwayFromZero), FloatingPointHelper<FloatingPointDimHelper>.Round(helperValue, MidpointRounding.AwayFromZero).Value);
```
Comment (tannergooding) [inline src/libraries/System.Runtime/tests/System/Numerics/IFloatingPointTests.cs:16]:
```suggestion
            Assert.Equal(float.Round(baseValue, MidpointRounding.AwayFromZero), FloatingPointHelper<FloatingPointDimHelper>.Round(helperValue, MidpointRounding.AwayFromZero).Value);
```

File: src/libraries/System.Runtime/tests/System/Numerics/IFloatingPointTests.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        }
+
+        [Fact]
+        public static void ToEvenRoundingTest()
+        {
+            Assert.Equal(Single.Round(baseValue, MidpointRounding.ToEven), FloatingPointHelper<FloatingPointDimHelper>.Round(helperValue, MidpointRounding.ToEven).Value);
```
Comment (tannergooding) [inline src/libraries/System.Runtime/tests/System/Numerics/IFloatingPointTests.cs:22]:
```suggestion
            Assert.Equal(float.Round(baseValue, MidpointRounding.ToEven), FloatingPointHelper<FloatingPointDimHelper>.Round(helperValue, MidpointRounding.ToEven).Value);
```

File: src/libraries/System.Runtime/tests/System/Numerics/IFloatingPointTests.cs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        }
+
+        [Fact]
+        public static void ToNegativeInfinityRoundingTest()
+        {
+            Assert.Equal(Single.Round(baseValue, MidpointRounding.ToNegativeInfinity), FloatingPointHelper<FloatingPointDimHelper>.Round(helperValue, MidpointRounding.ToNegativeInfinity).Value);
```
Comment (tannergooding) [inline src/libraries/System.Runtime/tests/System/Numerics/IFloatingPointTests.cs:28]:
```suggestion
            Assert.Equal(float.Round(baseValue, MidpointRounding.ToNegativeInfinity), FloatingPointHelper<FloatingPointDimHelper>.Round(helperValue, MidpointRounding.ToNegativeInfinity).Value);
```

File: src/libraries/System.Runtime/tests/System/Numerics/IFloatingPointTests.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        }
+
+        [Fact]
+        public static void ToPositiveRoundingTest()
+        {
+            Assert.Equal(Single.Round(baseValue, MidpointRounding.ToPositiveInfinity), FloatingPointHelper<FloatingPointDimHelper>.Round(helperValue, MidpointRounding.ToPositiveInfinity).Value);
```
Comment (tannergooding) [inline src/libraries/System.Runtime/tests/System/Numerics/IFloatingPointTests.cs:34]:
```suggestion
            Assert.Equal(float.Round(baseValue, MidpointRounding.ToPositiveInfinity), FloatingPointHelper<FloatingPointDimHelper>.Round(helperValue, MidpointRounding.ToPositiveInfinity).Value);
```

File: src/libraries/System.Runtime/tests/System/Numerics/IFloatingPointTests.cs [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        }
+
+        [Fact]
+        public static void ToZeroRoundingTest()
+        {
+            Assert.Equal(Single.Round(baseValue, MidpointRounding.ToZero), FloatingPointHelper<FloatingPointDimHelper>.Round(helperValue, MidpointRounding.ToZero).Value);
```
Comment (tannergooding) [inline src/libraries/System.Runtime/tests/System/Numerics/IFloatingPointTests.cs:40]:
```suggestion
            Assert.Equal(float.Round(baseValue, MidpointRounding.ToZero), FloatingPointHelper<FloatingPointDimHelper>.Round(helperValue, MidpointRounding.ToZero).Value);
```

**Review by tannergooding — APPROVED** [review]
Changes LGTM. Just a few minor nits to use `float` instead of using `Single`

**Review by dakersnar — APPROVED** [review]
LGTM, and thanks for submitting unit tests too.

**Pull request merged**